doc/reference/networking/promiscuous.rst

```diff
@@ -6,6 +6,59 @@ Promiscuous Mode
 Overview
 ********
 
+The network promiscuous API can be used to receive all network traffic
```

Comment:
Maybe add a definition of promiscuous mode:
```
Promiscuous mode is a mode for a network interface controller that
causes it to pass all traffic it receives to the application rather than
passing only the frames that the controller is specifically programmed
to receive. This mode is normally used for packet sniffing as used
to diagnose network connectivity issues by showing an application
all the data being transferred over the network.  (See the
`Wikipedia article on promiscuous mode 
<https://en.wikipedia.org/wiki/Promiscuous_mode>`_ for more information.)

The network promiscuous APIs are used to enable and disable this mode,
and to wait for and receive a network data to arrive. Not all network
technologies or network device drivers support promiscuous mode.  
```

Reply:
Sure, good idea.